controlpanel/api/kubernetes.py

```diff
@@ -1,9 +1,35 @@
+from copy import deepcopy
 import inspect
+import kubernetes
+import os
 
 from crequest.middleware import CrequestMiddleware
 from django.conf import settings
 
-from controlpanel.kubeapi.views import kubernetes, load_kube_config
+# This patch fixes incorrect base64 padding in the Kubernetes Python client.
+# Hopefully it will be fixed in the next release.
+from controlpanel.kubeapi import oidc_patch
+
+
+def get_config():
+    """
+    Load and returns a kubernetes Configuration instance
+    """
+
+    if "KUBERNETES_SERVICE_HOST" in os.environ:
+        kubernetes.config.load_incluster_config()
+    else:
+        kubernetes.config.load_kube_config()
+
+    config = kubernetes.client.Configuration()
+
+    # The deepcopy is to avoid a race conditions on the credentials keys
```

Comment:
Can you expand this comment a bit? Something like `A deepcopy of the configuration is used to avoid a race condition caused by subsequent calls to Configuration() reusing a singleton datastructure`

Reply:
👍